pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/BaseTaskExecutor.java

```diff
@@ -132,9 +140,19 @@ protected void downloadSegmentToLocal(String tableNameWithType, String segmentNa
           Collections.shuffle(uris);
           return uris;
         }, tarredSegmentFile, crypterName);
+        // untar the segment file
+        indexDir = TarCompressionUtils.untar(tarredSegmentFile, segmentDir).get(0);
       } else {
         throw e;
       }
+    } finally {
+      if (!FileUtils.deleteQuietly(tarredSegmentFile)) {
+        LOGGER.warn("Failed to delete tarred input segment: {}", tarredSegmentFile.getAbsolutePath());
+      }
+    }
+    if (indexDir == null) {
```

Comment:
Will it ever be `null`? Seems like it can never be `null`, or exception is thrown

Reply:
Hmm good point! I see in almost all scenarios exception been thrown or indexDir been non-null. Updated!